### PR TITLE
FLUX FP8 baseline implementation for MLPerf

### PR DESCRIPTION
**Pull request by zirui**
## Summary

- Add a baseline FLUX FP8 implementation for MLPerf.
- Add the corresponding configuration and launch examples.

## Validation

- Tested only on a single MI355X node.
- Validated against BF16 for numerical accuracy and convergence. FP8 validation loss has a small but consistent upward bias of approximately 2.2e-4 to 2.75e-4 relative to BF16.

## Status

This is a baseline implementation. Performance optimization and feature coverage are still in progress.

### CODE REVIEW

File: run_with_docker.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+DOCKER_IMAGE=${DOCKER_IMAGE:-rocm/primus:v26.2}
+CONTAINER_NAME=${CONTAINER_NAME:-primus-flux-mlperf}
+NOFILE_LIMIT=${NOFILE_LIMIT:-1048576}
+
+export GPUS_PER_NODE=${GPUS_PER_NODE:-8}
+export DATASET_PATH=${DATASET_PATH:-/data/cc12m_preprocessed}
+export EVAL_DATASET_PATH=${EVAL_DATASET_PATH:-/data/coco_preprocessed}
+export EMPTY_ENCODINGS_PATH=${EMPTY_ENCODINGS_PATH:-/data/empty_encodings}
+export OUTPUT_DIR=${OUTPUT_DIR:-/shared_nfs/zirui/runs/flux_rcp_compare/primus}
+export SEED=${SEED:-1234}
```
[the remark this answers is not in the document]
Comment (zirui) [reply run_with_docker.sh]: Fixed in 9234b974 by removing this broken duplicate root launcher. The supported portable launcher remains at `examples/mlperf/flux1/run_with_docker.sh`.

File: runner/helpers/hooks/train/pretrain/diffusion/requirements-diffusion.txt [hunk 2]
```diff
@@ -14,3 +14,5 @@ imageio[ffmpeg]
 wandb
 pydantic
 transformers==4.50.0
+git+https://github.com/mlcommons/logging.git@6.0.0-rc5
+git+https://github.com/NVIDIA/mlperf-common.git
```
[the remark this answers is not in the document]
Comment (zirui) [reply runner/helpers/hooks/train/pretrain/diffusion/requirements-diffusion.txt]: Fixed in 9234b974 by pinning mlperf-common to `b86d175a05849d650a8ff69c1e2c37b9f4e61d51`.

File: examples/mlperf/flux1/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+datasets>=3.6.0
+webdataset==1.0.2
+git+https://github.com/mlcommons/logging.git@6.0.0-rc5
+git+https://github.com/NVIDIA/mlperf-common.git
```
[the remark this answers is not in the document]
Comment (zirui) [reply examples/mlperf/flux1/requirements.txt]: Fixed in 9234b974 by pinning mlperf-common to `b86d175a05849d650a8ff69c1e2c37b9f4e61d51`.

File: primus/backends/diffusion/models/flux/layers.py [hunk 4]
```diff
@@ -53,20 +53,21 @@ def __init__(self, in_dim: int, hidden_dim: int):
...
-        self.scale = nn.Parameter(torch.ones(dim))
-
-    def forward(self, x: Tensor) -> Tensor:
-        x_dtype = x.dtype
-        x = x.float()
-        rrms = torch.rsqrt(torch.mean(x**2, dim=-1, keepdim=True) + 1e-6)
-        return (x * rrms).to(dtype=x_dtype) * self.scale
+        # Match TorchTitan's nn.RMSNorm(dim): eps remains None and PyTorch
+        # selects the epsilon for the BF16 input dtype at runtime.
+        super().__init__(dim)
```
[the remark this answers is not in the document]
Comment (zirui) [reply primus/backends/diffusion/models/flux/layers.py:70]: No change needed here: `torch.nn.RMSNorm` defines `eps=None` by default. Calling `super().__init__(dim)` therefore keeps `eps` as `None`, and PyTorch selects `torch.finfo(x.dtype).eps` at runtime. The comment describes that behavior accurately.

File: pyproject.toml [hunk 5]
```diff
@@ -48,6 +48,8 @@ dependencies = [
     "mlflow==3.11.1",
     "pyrsmi",
     "plotext",
+    "mlperf-logging @ git+https://github.com/mlcommons/logging.git@6.0.0-rc5",
+    "mlperf-common @ git+https://github.com/NVIDIA/mlperf-common.git",
```
[the remark this answers is not in the document]
Comment (zirui) [reply pyproject.toml]: Fixed in 9234b974 by pinning mlperf-common to `b86d175a05849d650a8ff69c1e2c37b9f4e61d51` across all dependency files.

File: primus/backends/diffusion/trainers/base.py [hunk 6]
```diff
@@ -287,12 +404,181 @@ def _setup_wandb(self):
...
+
+    def _mlperf_log_run_start(self):
+        if not self.mlperf_enabled:
+            return
+        if self.rank != 0:
+            return
+        c = self.mlperf_constants
+        opt = self.optimizer.param_groups[0]
+        self.mlperf_logger.event(key=c.CACHE_CLEAR, value="True")
+        self.mlperf_logger.event(key=c.SUBMISSION_BENCHMARK, value="flux1")
```
[the remark this answers is not in the document]
Comment (zirui) [reply primus/backends/diffusion/trainers/base.py:475]: Fixed in 9234b974. The CACHE_CLEAR value now follows `MLPERF_CLEAR_CACHES`, with a regression test for the disabled case.

**Review by wenxie-amd — DISMISSED** [review]

**Review by wenxie-amd — APPROVED** [review]
